packages/opentelemetry-exporter-zipkin/src/zipkin.ts

```diff
@@ -0,0 +1,171 @@
+import * as http from 'http';
+import * as https from 'https';
+import * as url from 'url';
+import { NoopLogger } from '@opentelemetry/core';
+import {
+  SpanExporter,
+  ReadableSpan,
+  ExportResult,
+} from '@opentelemetry/basic-tracer';
+
+import * as zipkinTypes from './types';
+import {
+  toZipkinSpan,
+  STATUS_CODE_TAG_NAME,
+  STATUS_DESCRIPTION_TAG_NAME,
+} from './transform';
+
+/**
+ * Zipkin Exporter
+ */
+export class ZipkinExporter implements SpanExporter {
+  static readonly DEFAULT_URL = 'http://localhost:9411/api/v2/spans';
+
+  private readonly _forceFlush: boolean;
+  private readonly _logger: types.Logger;
+  private readonly _serviceName: string;
+  private readonly _statusCodeTagName: string;
+  private readonly _statusDescriptionTagName: string;
+  private readonly _url: url.UrlWithStringQuery;
+
+  constructor(config: zipkinTypes.ExporterConfig) {
+    const urlStr = config.url || ZipkinExporter.DEFAULT_URL;
+
+    this._forceFlush = config.forceFlush || true;
+    this._logger = config.logger || new NoopLogger();
+    this._serviceName = config.serviceName;
+    this._statusCodeTagName = config.statusCodeTagName || STATUS_CODE_TAG_NAME;
+    this._statusDescriptionTagName =
+      config.statusDescriptionTagName || STATUS_DESCRIPTION_TAG_NAME;
+    this._url = url.parse(urlStr);
+  }
+
+  /**
+   * Export spans.
+   */
+  export(
+    spans: ReadableSpan[],
+    resultCallback: (result: ExportResult) => void
+  ) {
+    this._logger.debug('Zipkin exporter export');
+    // @todo: buffer spans (batch based on both time and max number)
+    return this._sendSpans(spans, resultCallback);
+  }
+
+  /**
+   * Shutdown exporter. Noop operation in this exporter.
+   */
+  shutdown() {
+    this._logger.debug('Zipkin exporter shutdown');
+    // Make an optimistic flush.
+    if (this._forceFlush) {
+      // @todo get spans from span processor (batch)
+      this._sendSpans([]);
+    }
+  }
+
+  /**
+   * Transforms an OpenTelemetry span to a Zipkin span.
+   */
+  private _toZipkinSpan(span: ReadableSpan): zipkinTypes.Span {
+    return toZipkinSpan(
+      span,
+      this._serviceName,
+      this._statusCodeTagName,
+      this._statusDescriptionTagName
+    );
+  }
+
+  /**
+   * Transform spans and sends to Zipkin service.
+   */
+  private _sendSpans(
+    spans: ReadableSpan[],
+    done?: (result: ExportResult) => void
+  ) {
+    const zipkinSpans = spans.map(span => this._toZipkinSpan(span));
+    return this._send(zipkinSpans, (result: ExportResult) => {
+      if (done) {
+        return done(result);
+      }
+    });
+  }
+
+  /**
+   * Send spans to the remote Zipkin service.
+   */
+  private _send(
+    zipkinSpans: zipkinTypes.Span[],
+    done: (result: ExportResult) => void
+  ) {
+    if (zipkinSpans.length === 0) {
+      this._logger.debug('Zipkin send with empty spans');
+      return done(ExportResult.Success);
+    }
+
+    const options = Object.assign(
```

Comment:
Does it make sense to do this in the constructor, instead of assigning every send request?